test/dataset.json

```diff
@@ -0,0 +1,7 @@
+{
+  // Elasticsearch index will be given this name.
```

Comment:
The JSON language does not support comments, can we remove them from here and document them in the README if necessary? That would eliminate the need to parse them in the indexer, which is pretty fragile as it doesn't support the full range of in-line comments.

Reply:
I think it's much easier -- for both Data explorer user and Data explorer developer -- if comments are directly in the JSON, vs having to switch back and forth between README and JSON. I don't think not supporting all possible comment formats is a problem. If people are going to add a comment, most likely they'll copy existing format.

Reply:
IIRC the indexer requires the name and primary_key to be present: since we've already documented what should be contained in this file elsewhere, adding the additional documentation here isn't strictly necessary. In fact, if I were a new user, it might be convenient for me to be able to have this information presented in a README so that I don't have to go digging through the config parsing code to figure out how my dataset should be structured.

If we think comments are going to be necessary for clarity here, I would do it this way:
- The JSON spec does not support comments, and this config will be un-parseable as JSON by any other tooling out there. Because of this, we should change the extension of this file from .json to .txt or something similar. We want to avoid bad surprises when my SomeTool.asJson('dataset.json') returns a failure.
- We should clearly document the JSON-based configuration language extended to support comments that is required by the dataset.txt file. Different languages have lots of different ways of representing comments, and we shouldn't assume that everybody will pick up that ^// in-line comments are the only supported comment format here.

As far as I can tell, Elasticsearch documents follow the JSON spec and do not have comments, in which case I would still suggest just dropping the comments, letting JSON be JSON, and documenting what a dataset.json should look like. Since this is based on the Elasticsearch document, it shouldn't be totally unfamiliar to anybody setting this up.

Reply:
To summarize offline discussion:

- I think it's much easier to have README say "cp -r config directory. Fill out config files according to comments.", rather than "Copy/paste this template for file 1. Here are 10 descriptions for 10 fields in file 1. Copy/paste this template for file 2....". Also, I think it's better for maintainability to have comments inline. Note the config files will become much more complicated over time.
- To support a variety of comment styles, I will use jsmin. This is how the JSON creator recommends handling JSON comments.